### ISSUE TITLE
Does not minify svgs that have the <use> tag in them

### ISSUE THREAD

**Issue by bencooper222** [posthtml/htmlnano#47]
Pretty much what it sounds like. I created an example [here](https://glitch.com/edit/#!/lying-equinox?path=server.js:24:4). You can view the results of this code in three places:

1. No minification: https://lying-equinox.glitch.me/
2. Minification with minifySvg set to `false`: https://lying-equinox.glitch.me/minify_good
3. Minification with minifySvg set to default (which is true): https://lying-equinox.glitch.me/minify_bad

If you inspect element the `minify_bad` path, you can see how htmlnano butchers the `<use>` tag's contents. Not sure if this is intentional behavior or not but it makes dealing with svgs difficult (which is unfortunate because they tend to create a lot of whitespace and such if not minimized).

**Comment by bencooper222** [comment]
The issue seems to be related to SVGO's default settings which seem to be too aggressive in minimizing.

**Comment by davidnagli** [comment]
@voischev @Scrum @maltsev We need this feature for @parcel-bundler. It's currently causing issues for our users. 😞 

See: parcel-bundler/parcel#929

**Comment by Scrum** [comment]
@bencooper222 @davidnagli Thanks for ping, I'll try to figure this out soon. Let me know if you find a solution ahead of me :)

**Comment by Scrum** [comment]
At the moment I recommend using htmlnano with settings for svgo:
```js
minifySvg: {
    plugins: [ 
        { cleanupIDs: false },
        { removeUnknownsAndDefaults: false},
    ]
}
```

see [example](https://github.com/inside-demo/svgo-demo)

> The issue seems to be related to SVGO's default settings which seem to be too aggressive in minimizing.

yep

**Comment by Scrum** [comment]
related to work https://github.com/svg/svgo/issues/921

**Comment by bfred-it** [comment]
In my case, the entire `<svg>` is replaced with `undefined` if a `use` tag is in it:

For example
```html
<!DOCTYPE html><svg><use xlink:href="./file.svg"></svg>
```

Becomes

```html
<!DOCTYPE html>undefined
```

Here's a one-line CLI test:

```sh
$node -e 'require("htmlnano").process(\'<!DOCTYPE html><svg><use xlink:href="./file.svg"></svg>\').then(result => console.log(result.html));'

<!DOCTYPE html>undefined
```

**Comment by maltsev** [comment]
I updated all dependencies and released [version 0.1.7](https://github.com/posthtml/htmlnano/releases/tag/0.1.7). Now `<use>` minification works well.

**Comment by maltsev** [comment]
@bfred-it `xlink:href` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href). Therefore, I guess, SVGO doesn't process it. If you replace it with `href` it'd work fine, though.

**Comment by bfred-it** [comment]
@maltsev `xlink:href` should still be processed because the vast majority of SVGs on the web use it. I doubt `href` has the same compat. 

I think my issue here was the lack of `xmlns:xlink="http://www.w3.org/1999/xlink"`

This works:

```html
<!DOCTYPE html><svg xmlns:xlink="http://www.w3.org/1999/xlink"><use xlink:href="./file.svg"/></svg>
```

This doesn't, but it should **fail** either explicitly or by leaving the SVG as-is, not output `undefined`
```html
<!DOCTYPE html><svg><use xlink:href="./file.svg"/></svg>
```

**Comment by maltsev** [comment]
> I think my issue here was the lack of xmlns:xlink="http://www.w3.org/1999/xlink"

You're right. `htmlnano` processes such code correctly:
```js
require('htmlnano')
    .process('<!DOCTYPE html><svg xmlns:xlink="http://www.w3.org/1999/xlink"><use xlink:href="./file.svg"/></svg>')
    .then(result => console.log(result.html));

// Output:
// <!DOCTYPE html><svg xmlns:xlink="http://www.w3.org/1999/xlink"><use xlink:href="./file.svg"/></svg>
```

But it explicitly fails when `xmlns:xlink` isn't present:
```js
require("htmlnano")
    .process('<!DOCTYPE html><svg><use xlink:href="./file.svg"/></svg>')
    .then(result => console.log(result.html))
    .catch(err => console.error(err));

// Output:
// Error in parsing SVG: Unbound namespace prefix: "xlink"
// Line: 0
// Column: 34
// Char: >
```

So it seems that that `undefined` error should be fixed in parcel-bundler/parcel#996.
